# Instagram Private API

In [46]:
# imports
from instagrapi import Client


In [47]:
# account keys
ACCOUNT_USERNAME = 'ringledingleshingle'
ACCOUNT_PASSWORD = 'nobatteriesinvolved!'

In [48]:
# account login
cl = Client()
cl.login(ACCOUNT_USERNAME, ACCOUNT_PASSWORD)
print('login complete')

login complete


In [58]:
# get personal account info
cl.account_info().dict()
# can also change things like bio, name, etc. with cl.account_edit()
# docs: https://adw0rd.github.io/instagrapi/usage-guide/account.html

{'pk': '60484357702',
 'username': 'ringledingleshingle',
 'full_name': 'Ringle Dingle',
 'is_private': False,
 'profile_pic_url': HttpUrl('https://scontent-yyz1-1.cdninstagram.com/v/t51.2885-19/44884218_345707102882519_2446069589734326272_n.jpg?_nc_ht=scontent-yyz1-1.cdninstagram.com&_nc_cat=1&_nc_ohc=kK4ty5pcIEgAX8Xgl28&edm=AAAAAAABAAAA&ccb=7-5&ig_cache_key=YW5vbnltb3VzX3Byb2ZpbGVfcGlj.2-ccb7-5&oh=00_AfDBqePY7M1hBFs1MZjpZTdSLe8dqusbZmMv8qOVOId4Lw&oe=649B978F', ),
 'is_verified': False,
 'biography': '',
 'external_url': None,
 'is_business': False,
 'birthday': '2000-06-23',
 'phone_number': '',
 'gender': 3,
 'email': 'theeshingle@gmail.com'}

In [59]:
# get user id based on username
TARGET = 'robertpatel21'#'hannahta0' #
user_id = cl.user_id_from_username(TARGET)
print(user_id)
#medias = cl.user_medias(cl.user_id, 20)

5946131059


In [60]:
# get acc media
medias = cl.user_medias(user_id, 20)
print(medias)
# write medias to a file
with open('medias.txt', 'w') as f:
    f.write(str(medias))
f.close()

[Media(pk='2953153950610174008', id='2953153950610174008_5946131059', code='Cj7td8hKTA4', taken_at=datetime.datetime(2022, 10, 20, 10, 56, 46, tzinfo=datetime.timezone.utc), media_type=1, image_versions2={}, product_type='', thumbnail_url=HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-15/312330535_1173543736843601_5459429612216261578_n.webp?stp=dst-jpg_e35&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=105&_nc_ohc=GmZPT24Ma6MAX9Cr-5B&edm=APU89FABAAAA&ccb=7-5&oh=00_AfBJdP_ZXT0icktq2y0PnmFZ7q8AIwdMlVOFYTfvN7SvWA&oe=649A560B&_nc_sid=f4eaf9', ), location=None, user=UserShort(pk='5946131059', username='robertpatel21', full_name='', profile_pic_url=None, profile_pic_url_hd=None, is_private=None, stories=[]), comment_count=0, comments_disabled=False, commenting_disabled_for_viewer=False, like_count=69, play_count=None, has_liked=None, caption_text='', accessibility_caption=None, usertags=[], sponsor_tags=[], video_url=None, view_count=0, video_duration=0.0, title='', resources=

In [61]:
# convert medias entries to dict entries
medias_dict = [medias[i].dict() for i in range(len(medias))]
print(medias_dict)

[{'pk': '2953153950610174008', 'id': '2953153950610174008_5946131059', 'code': 'Cj7td8hKTA4', 'taken_at': datetime.datetime(2022, 10, 20, 10, 56, 46, tzinfo=datetime.timezone.utc), 'media_type': 1, 'image_versions2': {}, 'product_type': '', 'thumbnail_url': HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-15/312330535_1173543736843601_5459429612216261578_n.webp?stp=dst-jpg_e35&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=105&_nc_ohc=GmZPT24Ma6MAX9Cr-5B&edm=APU89FABAAAA&ccb=7-5&oh=00_AfBJdP_ZXT0icktq2y0PnmFZ7q8AIwdMlVOFYTfvN7SvWA&oe=649A560B&_nc_sid=f4eaf9', ), 'location': None, 'user': {'pk': '5946131059', 'username': 'robertpatel21', 'full_name': '', 'profile_pic_url': None, 'profile_pic_url_hd': None, 'is_private': None, 'stories': []}, 'comment_count': 0, 'comments_disabled': False, 'commenting_disabled_for_viewer': False, 'like_count': 69, 'play_count': None, 'has_liked': None, 'caption_text': '', 'accessibility_caption': None, 'usertags': [], 'sponsor_tags': [], 'vi

In [62]:
# what keys are avaialable
print(medias_dict[0].keys())

dict_keys(['pk', 'id', 'code', 'taken_at', 'media_type', 'image_versions2', 'product_type', 'thumbnail_url', 'location', 'user', 'comment_count', 'comments_disabled', 'commenting_disabled_for_viewer', 'like_count', 'play_count', 'has_liked', 'caption_text', 'accessibility_caption', 'usertags', 'sponsor_tags', 'video_url', 'view_count', 'video_duration', 'title', 'resources', 'clips_metadata'])


In [63]:
# Hashtag search

hashtag = cl.hashtag_info('groypers')
print(hashtag.dict())

{'id': '18009227209031895', 'name': 'groypers', 'media_count': 1700, 'profile_pic_url': HttpUrl('https://scontent-lga3-1.cdninstagram.com/v/t51.2885-15/82281768_485338449087423_7712361184840834090_n.jpg?stp=dst-jpg_e35_s150x150&_nc_ht=scontent-lga3-1.cdninstagram.com&_nc_cat=103&_nc_ohc=jj49QGXxh0MAX9KFHrI&edm=ABZsPhsBAAAA&ccb=7-5&oh=00_AfC6bqBVwztq9kQV9dj4OlUBRTRW274ai0LbgPfUDlWDCA&oe=649A4742&_nc_sid=e1d802', )}


In [51]:
# top posts 

top = cl.hashtag_medias_top('groypers', 9)
print(top)



[Media(pk='3130988664497612354', id='3130988664497612354_30104905256', code='CtzgZoHJ6ZC', taken_at=datetime.datetime(2023, 6, 22, 19, 42, 36, tzinfo=datetime.timezone.utc), media_type=1, image_versions2={}, product_type='', thumbnail_url=HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-15/355661474_218766887750057_932615320587936737_n.jpg?stp=dst-jpg_e35&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=105&_nc_ohc=QB8encIB-YcAX-5iEVc&edm=AP_V10EBAAAA&ccb=7-5&oh=00_AfDFyBI6HelyUL2aH6N89iWMxkYbvRyaudFIDi61ZWQ8Ew&oe=649A64B8&_nc_sid=8721cf', ), location=None, user=UserShort(pk='30104905256', username='smol_chungus_', full_name='Smol Chungus', profile_pic_url=HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-19/88397673_1247885145407099_3106625625599770624_n.jpg?stp=dst-jpg_s150x150&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=111&_nc_ohc=vEo_0sOZnMYAX_LE_Xg&edm=AP_V10EBAAAA&ccb=7-5&oh=00_AfDZDICuTdREcfMObN9k2h1FBmS8CkPGXCyFjH4Ko6eahQ&oe=649A8766&_nc_sid=8721cf', 

In [54]:
print(top[0].dict())


{'pk': '3130988664497612354', 'id': '3130988664497612354_30104905256', 'code': 'CtzgZoHJ6ZC', 'taken_at': datetime.datetime(2023, 6, 22, 19, 42, 36, tzinfo=datetime.timezone.utc), 'media_type': 1, 'image_versions2': {}, 'product_type': '', 'thumbnail_url': HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-15/355661474_218766887750057_932615320587936737_n.jpg?stp=dst-jpg_e35&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=105&_nc_ohc=QB8encIB-YcAX-5iEVc&edm=AP_V10EBAAAA&ccb=7-5&oh=00_AfDFyBI6HelyUL2aH6N89iWMxkYbvRyaudFIDi61ZWQ8Ew&oe=649A64B8&_nc_sid=8721cf', ), 'location': None, 'user': {'pk': '30104905256', 'username': 'smol_chungus_', 'full_name': 'Smol Chungus', 'profile_pic_url': HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-19/88397673_1247885145407099_3106625625599770624_n.jpg?stp=dst-jpg_s150x150&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=111&_nc_ohc=vEo_0sOZnMYAX_LE_Xg&edm=AP_V10EBAAAA&ccb=7-5&oh=00_AfDZDICuTdREcfMObN9k2h1FBmS8CkPGXCyFjH4Ko6eahQ&oe

In [ ]:
def lst_scrape(lst):
    # lst is a list of dictionaries
    lst = elem_to_dict(lst)
    

{'pk': '3130988664497612354', 'id': '3130988664497612354_30104905256', 'code': 'CtzgZoHJ6ZC', 'taken_at': datetime.datetime(2023, 6, 22, 19, 42, 36, tzinfo=datetime.timezone.utc), 'media_type': 1, 'image_versions2': {}, 'product_type': '', 'thumbnail_url': HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-15/355661474_218766887750057_932615320587936737_n.jpg?stp=dst-jpg_e35&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=105&_nc_ohc=QB8encIB-YcAX-5iEVc&edm=AP_V10EBAAAA&ccb=7-5&oh=00_AfDFyBI6HelyUL2aH6N89iWMxkYbvRyaudFIDi61ZWQ8Ew&oe=649A64B8&_nc_sid=8721cf', ), 'location': None, 'user': {'pk': '30104905256', 'username': 'smol_chungus_', 'full_name': 'Smol Chungus', 'profile_pic_url': HttpUrl('https://scontent-lga3-2.cdninstagram.com/v/t51.2885-19/88397673_1247885145407099_3106625625599770624_n.jpg?stp=dst-jpg_s150x150&_nc_ht=scontent-lga3-2.cdninstagram.com&_nc_cat=111&_nc_ohc=vEo_0sOZnMYAX_LE_Xg&edm=AP_V10EBAAAA&ccb=7-5&oh=00_AfDZDICuTdREcfMObN9k2h1FBmS8CkPGXCyFjH4Ko6eahQ&oe=649A8766&_nc_sid=8721cf', ), 'profile_pic_url_hd': None, 'is_private': False, 'stories': []}, 'comment_count': 0, 'comments_disabled': False, 'commenting_disabled_for_viewer': False, 'like_count': 27, 'play_count': None, 'has_liked': None, 'caption_text': 'Prove me right in the comments 👇…\n.\n.\nIf you enjoyed this meme go ahead and give @smol_chungus_ a follow please.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n.\n#zuccmemes #rickyisms #trailerparkboysmemes #smolchungusmemes #tiktok #samhyde #animememes 💙 #chungus #pewdiepie #pewdiepiememes #pewdiepiesubmissions #samhydememes #followforfollowback #follow4followback #follow4like #zoomermemes #mrbeast #disneymemes #pinkguymemes #filthyfrank #groypers #maxmoefoe #jojimemes #letsgobrandon #groyperarmy #groyperseason #groypermemes #joerogan #fishtanklive #chungusmemes', 'accessibility_caption': None, 'usertags': [], 'sponsor_tags': [], 'video_url': None, 'view_count': 0, 'video_duration': 0.0, 'title': '', 'resources': [], 'clips_metadata': {}}

In [55]:
def elem_to_dict(lst):
    # lst is a list of Media objects
    return [lst[i].dict() for i in range(len(lst))]

In [ ]:
def caption_csv(lst):
    # lst is a list of dictionaries
    # write captions to a csv file


In [10]:
cl.user_followers_(cl.user_id).keys()

AttributeError: 'list' object has no attribute 'keys'